File: src/main/java/fr/ayfri/minecraft_art/buttons/SaveButton.java
package fr.ayfri.minecraft_art.buttons;

import fr.ayfri.minecraft_art.Main;

import java.io.File;

public class SaveButton extends Button {
	public SaveButton(final Main sketch) {
		super(sketch, sketch.width / 2 + 200, sketch.height - 100, 150, 80);
		setText("Save");
	}
	
	@Override
	public void onClick() {
		final File defaultSelectedFile = new File("image.png");
		sketch.selectOutput("Select where to save the image.", "save", defaultSelectedFile);
	}
}

File: src/main/kotlin/io/github/ayfri/minecraft_art/buttons/SaveButton.kt
package io.github.ayfri.minecraft_art.buttons

import io.github.ayfri.minecraft_art.Main
import java.io.File

class SaveButton(sketch: Main) : Button(sketch, sketch.width / 2 + 200, sketch.height - 100, 150, 80) {
	init {
		text = "Save"
	}

	override fun onClick() = sketch.selectOutput("Select where to save the image.", "save", File("image.png"))
}
